src/MultiOwnerLightAccount.sol

```diff
@@ -0,0 +1,336 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.23;
+
+import {IERC1271} from "@openzeppelin/contracts/interfaces/IERC1271.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {MessageHashUtils} from "@openzeppelin/contracts/utils/cryptography/MessageHashUtils.sol";
+import {SignatureChecker} from "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
+import {BaseAccount} from "account-abstraction/core/BaseAccount.sol";
+import {SIG_VALIDATION_FAILED} from "account-abstraction/core/Helpers.sol";
+import {IEntryPoint} from "account-abstraction/interfaces/IEntryPoint.sol";
+import {PackedUserOperation} from "account-abstraction/interfaces/PackedUserOperation.sol";
+import {TokenCallbackHandler} from "account-abstraction/samples/callback/TokenCallbackHandler.sol";
+import {CastLib} from "modular-account/helpers/CastLib.sol";
+import {SetValue} from "modular-account/libraries/Constants.sol";
+import {LinkedListSet, LinkedListSetLib} from "modular-account/libraries/LinkedListSetLib.sol";
+
+import {CustomSlotInitializable} from "./CustomSlotInitializable.sol";
+
+/// @title A simple ERC-4337 compatible smart contract account with one or more designated owner accounts.
+/// @dev Like LightAccount, but multiple owners are supported. The account is initialized with a list of owners,
+/// and the `updateOwners` method can be used to add or remove owners.
+contract MultiOwnerLightAccount is
+    BaseAccount,
+    TokenCallbackHandler,
+    UUPSUpgradeable,
+    CustomSlotInitializable,
+    IERC1271
+{
+    using ECDSA for bytes32;
+    using MessageHashUtils for bytes32;
+    using LinkedListSetLib for LinkedListSet;
+    using CastLib for address;
+    using CastLib for SetValue[];
+
+    // keccak256(abi.encode(uint256(keccak256("multi_owner_light_account_v1.storage")) - 1)) & ~bytes32(uint256(0xff));
+    bytes32 internal constant _STORAGE_POSITION = 0x0eb5184329babcda7203727c83eff940fb292fc735f61720a6182b755bf7f900;
+    // keccak256(abi.encode(uint256(keccak256("multi_owner_light_account_v1.initializable")) - 1)) & ~bytes32(uint256(0xff));
```

Comment:
I know this is "v1 of multi owner LA", but would it make sense to bump this to v2 to line up with the light account versioning?

Reply:
I was thinking about this a bit and thinking it might be nice to disassociate storage versions from implementation versions, since storage might not always need to change when account interfaces change. But I also see how that could get confusing.

I'm thinking about keeping the original light account on `light_account_v1.storage` too. Thoughts?

Reply:
I see, yes it makes sense to have separate storage versioning numbers. And it also makes sense to keep LA v2's storage at the V1 position, because the format hasn't changed.